sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc

```diff
@@ -0,0 +1,539 @@
+struct is_property_key_of<
+  sycl::ext::intel::experimental::awidth_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::dwidth_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::read_write_mode_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::latency_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::maxburst_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::wait_request_key,
+  annotated_ptr<T, PropertyListT>> : std::true_type {};
+
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::conduit_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::register_map_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::stable_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::buffer_location_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::awidth_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::dwidth_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::read_write_mode_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::latency_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::maxburst_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+  sycl::ext::intel::experimental::wait_request_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+} // namespace sycl::ext::oneapi::experimental
+```
+--
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
```

Comment:
I have a general comment about all these properties.  What is the behavior when the kernel is submitted to a non-FPGA device?  I think there are two options:

* They are ignored unless the device is FPGA, or
* They are "optional kernel features", which means that the user is responsible for checking the device type before submitting the kernel.  If the user makes a mistake, the `parallel_for` call raises a synchronous exception.

I think we choose the first option for some of the other FPGA properties.

Reply:
Added a note stating (1) in the overview section.

Reply:
Thanks.  See my other comment, though, about stating this in a more normative way.